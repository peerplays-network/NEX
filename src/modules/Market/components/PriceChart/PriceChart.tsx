import dynamic from "next/dynamic";

const DynamicPriceChart = dynamic(() => import("./CreatePriceChart"), {
  ssr: false,
});

export const PriceChart = (): JSX.Element => {
  return <DynamicPriceChart />;
};
