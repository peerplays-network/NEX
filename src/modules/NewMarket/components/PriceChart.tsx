import { createChart } from "lightweight-charts";
import React, { useEffect, useRef } from "react";

export const PriceChart = (): JSX.Element => {
  const chartContainerRef = useRef();

  useEffect(() => {
    if (chartContainerRef.current) {
      const chart = createChart(chartContainerRef.current, {
        width: 1000,
        height: 600,
        layout: {
          textColor: "black",
          background: { type: "solid", color: "white" },
        },
        grid: {
          vertLines: {
            color: "rgba(197, 203, 206, 0.5)",
          },
          horzLines: {
            color: "rgba(197, 203, 206, 0.5)",
          },
        },

        rightPriceScale: {
          borderColor: "rgba(197, 203, 206, 0.8)",
        },
        timeScale: {
          borderColor: "rgba(197, 203, 206, 0.8)",
        },
      });

      const candleSeries = chart.addCandlestickSeries({
        upColor: "#26a69a",
        downColor: "#ef5350",
        wickDownColor: "#ef5350",
        wickUpColor: "#26a69a",
        borderVisible: false,
      });

      candleSeries.setData([
        {
          time: "2018-10-19",
          open: 180.34,
          high: 180.99,
          low: 178.57,
          close: 179.85,
        },
        {
          time: "2018-10-22",
          open: 180.82,
          high: 181.4,
          low: 177.56,
          close: 178.75,
        },
      ]);

      return () => {
        chart.remove();
      };
    }
  }, []);

  return <div ref={chartContainerRef} />;
};
