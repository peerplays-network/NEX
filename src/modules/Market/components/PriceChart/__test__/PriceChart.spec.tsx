import { cleanup, render } from "@testing-library/react";
import React from "react";

import { PriceChart } from "../PriceChart";

describe("PriceChart component", () => {
  afterEach(() => {
    cleanup();
  });

  it("should render the component correctly", () => {
    const { container, asFragment } = render(<PriceChart />);

    expect(asFragment()).toMatchSnapshot();

    expect(container.firstChild).toBeDefined();
  });
});
