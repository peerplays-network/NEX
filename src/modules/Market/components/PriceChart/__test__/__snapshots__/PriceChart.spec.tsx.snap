// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceChart component should render the component correctly 1`] = `<DocumentFragment />`;
